### ISSUE TITLE
Sometimes the `$removeparam=` parameter does not work

### ISSUE THREAD

**Issue by shenzhiming88** [uBlockOrigin/uBlock-issues#1767]
### Prerequisites

- [X] I performed a [cursory search of the issue tracker](https://github.com/uBlockOrigin/uBlock-issues/issues?q=is%3Aissue) to avoid opening a duplicate issue
- [X] The issue is not present after wholly disabling uBlock Origin ("uBO") in the browser
- [X] I checked the [documentation](https://github.com/gorhill/uBlock/wiki) to understand that the issue I report is not a normal behavior

### I tried to reproduce the issue when...

- [X] uBO is the only extension
- [X] using a new, unmodified browser profile

### Description

If the page is not reloaded and the link in the address bar is changed, the parameters in the address bar are not automatically cleared.

### A specific URL where the issue occurs

https://www.baidu.com/s?wd=Test&rsv_spt=1&rsv_iqid=0xa83b935b000249f8&issp=1&f=8&rsv_bp=1&rsv_idx=2&ie=utf-8&rqlang=cn&tn=baiduhome_pg&rsv_enter=0&rsv_dl=tb&oq=Test&rsv_btype=t&rsv_t=01c48EfWFiNRaqUEe4zGwXwnOqY5SawI8LatW4wHB3AAVN9s5wUo2c27lp0&rsv_pq=bdd8c98800097d0c

### Steps to Reproduce

- Add the following rules
```
||baidu.com^$removeparam=/^rsv/
||baidu.com^$removeparam=cl
||baidu.com^$removeparam=f
||baidu.com^$removeparam=fr
||baidu.com^$removeparam=hisfilter
||baidu.com^$removeparam=inputT
||baidu.com^$removeparam=issp
||baidu.com^$removeparam=n_type
||baidu.com^$removeparam=oq
||baidu.com^$removeparam=p_from
||baidu.com^$removeparam=prefixsug
||baidu.com^$removeparam=rsp
||baidu.com^$removeparam=tn
```
- Open this page
- Click the search button in the search box
![2021-10-17_09-53-25](https://user-images.githubusercontent.com/66902050/137607430-08913484-a0c4-4eb4-8458-b58d30367058.gif)

### Expected behavior

The first time you open it, the page parameters are cleared. But after clicking the search button in the search bar, the page parameters will not be cleared.

### Actual behavior

Web parameters should be cleaned whenever.

### Configuration

_No need_

**Comment by uBlock-user** [comment]
`$removeparam` will only work on network requests, location url changed without firing a network request won't have params removed.

**Comment by gorhill** [comment]
Yes, duplicate of #1704. I verified with AdGuard, and the behavior is the same as with uBO. However, I can understand the confusion and I will investigate whether a somewhat trivial fix is possible before closing.

**Comment by shenzhiming88** [comment]
> > shenzhiming88 :
> > ### Prerequisites
> > ```
> >  [x]  I performed a cursory search of the issue tracker to avoid opening a duplicate issue
> > ```
> 
> Really? A duplicate of #1704

I'm sorry that since I'm not good with English, I didn't search for a similar question.
